### ISSUE TITLE
Fix users_db_security.js delete/recreate _users DB

### ISSUE THREAD

**Issue by wohali** [apache/couchdb#1238]
We got a random failure in Jenkins related to encryption:

https://builds.apache.org/blue/organizations/jenkins/CouchDB/detail/master/223/pipeline

Logs at https://couchdb-vm2.apache.org/ci_errorlogs/jenkins-couchdb-223-2018-03-23T23:15:34.663810/couchlog.tar.gz

```
test/javascript/tests/users_db_security.js                     
    Error: undefined
Trace back (most recent call first):
    
  52: test/javascript/test_setup.js
      T((void 0))
  35: test/javascript/tests/users_db_security.js
      ("jchris1")
  39: test/javascript/tests/users_db_security.js
      ([object Object],"org.couchdb.user:fdmanana","jchris1")
 168: test/javascript/tests/users_db_security.js
      ("pbkdf2",(function (derived_key) {TEquals(40, derived_key.length, "de
 413: test/javascript/tests/users_db_security.js
      ()
 395: test/javascript/couch_test_runner.js
      run_on_modified_server([object Array],(function () {try {testFun(schem
 411: test/javascript/tests/users_db_security.js
      ("pbkdf2",0,[object Array])
 399: test/javascript/tests/users_db_security.js
      ()
  37: test/javascript/cli_runner.js
      runTest()
  48: test/javascript/cli_runner.js
```

@eiri says:
> I looked into it and as far as I can see with bcrypt PR API test users_db_security.js was changed to be prune to racing condition - user database [got deleted and re-created](https://github.com/apache/couchdb/blob/master/test/javascript/tests/users_db_security.js#L414), which is known anti-pattern.

**Comment by eiri** [comment]
Taking this for a fix, it breaks every third travis run. 😠

**Comment by wohali** [comment]
Thanks @eiri, was going to get to it tomorrow if you didn't.

**Comment by janl** [comment]
I’m having a stab at it at the moment.

**Comment by eiri** [comment]
@janl ok, I'll stop my work and switch to something different. Was self-assignment on this case not sufficient to indicate I'm already working on it?

**Comment by janl** [comment]
@eiri sorry, I thought your clock ran out and the end of your day yesterday :/

**Comment by janl** [comment]
@eiri I felt a bit over-eager because I had approved this PR and I should have caught the issue on review.

**Comment by janl** [comment]
Closing this in favour of #1250. Apologies again for the toe-stepping.

**Comment by eiri** [comment]
@janl No worries, it's actually my bias of assuming me been online miracly obvious to everyone and I do seat in a weird timezone. Sorry this came across that grouchy.

**Comment by janl** [comment]
All good :)

**Comment by wohali** [comment]
@janl Your fix failed on Jenkins, reopening :(

https://builds.apache.org/blue/organizations/jenkins/CouchDB/detail/master/231/pipeline

```
test/javascript/tests/users_db_security.js                     
    Error: undefined
Trace back (most recent call first):
    
  52: test/javascript/test_setup.js
      T((void 0))
  35: test/javascript/tests/users_db_security.js
      ("jchris")
  39: test/javascript/tests/users_db_security.js
      ([object Object],"org.couchdb.user:jchris","jchris")
 111: test/javascript/tests/users_db_security.js
      ("bcrypt",(function (derived_key) {TEquals(60, derived_key.length, "de
 413: test/javascript/tests/users_db_security.js
      ()
 395: test/javascript/couch_test_runner.js
      run_on_modified_server([object Array],(function () {try {testFun(schem
 411: test/javascript/tests/users_db_security.js
      ("bcrypt",1,[object Array])
 399: test/javascript/tests/users_db_security.js
      ()
  37: test/javascript/cli_runner.js
      runTest()
  48: test/javascript/cli_runner.js
      
fail
```

https://couchdb-vm2.apache.org/ci_errorlogs/jenkins-couchdb-231-2018-03-28T17:21:43.727450/couchlog.tar.gz

**Comment by janl** [comment]
on it

**Comment by janl** [comment]
Two points:
1. we are in dire need of rewriting this test file
2. it *might* be that this failure points to pbkdf not working as we expect it to for config admins. I’ll keep investigating.

**Comment by eiri** [comment]
1. Totally agree, this test and the whole javascript suite are **really** hard to follow. It'd be amazing to revive Russel's suggestion to migrate it to Elixir.

2. I think it's not that `pbkdf` or `bcrypt` don't work as expected, but rather hashing on heavily restricted CI hosts is not able to complete [within 5 sec](https://github.com/apache/couchdb/blob/master/test/javascript/tests/users_db_security.js#L110)

In the logs from the failed jenkins' test I found:
```
[notice] 2018-03-28T17:21:33.282844Z node1@127.0.0.1 <0.32515.3> a82be2eaf1 127.0.0.1:15984 127.0.0.1 undefined PUT /_users/org.couchdb.user%3Ajchris 201 ok 511
[warning] 2018-03-28T17:21:38.285265Z node1@127.0.0.1 <0.32515.3> 59932a5bfc couch_httpd_auth: Authentication failed for user jchris from 127.0.0.1
[notice] 2018-03-28T17:21:38.285724Z node1@127.0.0.1 <0.32515.3> 59932a5bfc 127.0.0.1:15984 127.0.0.1 undefined POST /_session 401 ok 2
```

And when I insert something like
```javascript
    if (scheme == "bcrypt") {
      wait(500);
    }
    else {
      wait(5000);
    }
```
at line 110 of `users_db_security.js` I'm getting the same failure and stack trace.

**Comment by janl** [comment]
s/pbkdf/bcrypt/ apologies

**Comment by janl** [comment]
This is it, we need to send log_rounds to 1 as well

**Comment by janl** [comment]
Fixed in https://github.com/apache/couchdb/pull/1258
